plugins/inputs/dpdk/dpdk_utils.go

```diff
@@ -0,0 +1,123 @@
+// +build linux
+
+package dpdk
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+)
+
+func commandWithParams(command string, params string) string {
+	if params != "" {
+		return command + "," + params
+	}
+	return command
+}
+
+func stripParams(command string) string {
+	index := strings.IndexRune(command, ',')
+	if index == -1 {
+		return command
+	}
+	return command[:index]
+}
+
+// Since DPDK is an open-source project, developers can use their own format of params
+// so it could "/command,1,3,5,123" or "/command,userId=1, count=1234".
+// To avoid issues with different formats of params, all params are returned as single string
+func getParams(command string) string {
+	index := strings.IndexRune(command, ',')
+	if index == -1 {
+		return ""
+	}
+	return command[index+1:]
+}
+
+// Checks if provided path points to socket
+func isSocket(path string) error {
+	pathInfo, err := os.Lstat(path)
+	if os.IsNotExist(err) {
+		return fmt.Errorf("provided path does not exist: '%v'", path)
+	}
+
+	if err != nil {
+		return fmt.Errorf("cannot get system information of '%v' file: %v", path, err)
+	}
+
+	if pathInfo.Mode()&os.ModeSocket != os.ModeSocket {
+		return fmt.Errorf("provided path does not point to a socket file: '%v'", path)
+	}
+
+	return nil
+}
+
+// Converts JSON array containing devices identifiers from DPDK response to string slice
+func jsonToArray(input []byte, command string) ([]string, error) {
+	if len(input) == 0 {
+		return nil, fmt.Errorf("got empty object instead of json")
+	}
+
+	var rawMessage map[string]json.RawMessage
+	err := json.Unmarshal(input, &rawMessage)
+	if err != nil {
+		return nil, err
+	}
+
+	var intArray []int
+	var stringArray []string
+	err = json.Unmarshal(rawMessage[command], &intArray)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshall json response - %v", err)
+	}
+
+	for _, value := range intArray {
+		stringArray = append(stringArray, fmt.Sprintf("%v", value))
```

Comment:
To be honest I would prefer an explicit conversion like `strconv.FormatInt` or similar here instead of relying on automatic type conversion through `%v`...

Reply:
Good point, changed.